src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/UsersController.cs

```diff
@@ -15,11 +16,15 @@ namespace NHSD.BuyingCatalogue.Organisations.Api.Controllers
     [Produces("application/json")]
     public sealed class UsersController : Controller
     {
+        private readonly IRegistrationService _registrationService;
         private readonly IUsersRepository _usersRepository;
 
-        public UsersController(IUsersRepository usersRepository)
+        public UsersController(
+            IUsersRepository usersRepository,
+            IRegistrationService registrationService)
         {
             _usersRepository = usersRepository ?? throw new ArgumentNullException(nameof(usersRepository));
+            _registrationService = registrationService ?? throw new ArgumentNullException(nameof(registrationService));
```

Comment:
We have a cool .ThrowIfNull() extension that is sporadically used, and could maybe be used here

Reply:
I'm not a fan of `.ThrowIfNull()`.

While it is obviously syntactically fine semantically it is wrong to me – it shouldn't be possible to invoke a method on a null object – although we can due to the way extension methods are implemented.

I also think extension methods that essentially extend everything just smell – Mr Lippert had some [comments](https://stackoverflow.com/a/7652359) on this a few years back.

This approach also means that the call stack shows the exception occurring in the correct method rather than an extension method – admittedly this is a small difference in this case but another reason I prefer the version above.

Perhaps another one for the next code standards meeting...